### DISCUSSION TITLE
How to throw exceptions on bad logging configuration?

### DISCUSSION THREAD

**Discussion by makrushin-evgenii** [open-telemetry/opentelemetry-dotnet#5579] [Q&A]
Hi. Using packages `OpenTelemetry.Exporter.[OpenTelemetryProtocol|Console]:1.7.0`, I configure `LoggerFactory`:
```csharp
using Microsoft.Extensions.Logging;
using OpenTelemetry.Exporter;
using OpenTelemetry.Logs;
using OpenTelemetry.Resources;


var loggerFactory = LoggerFactory.Create(builder => builder
    .AddOpenTelemetry(options => options
        .AddConsoleExporter()
        .AddOtlpExporter(exporterOptions =>
        {
            exporterOptions.Endpoint = new Uri("https://***/v1/logs");
            exporterOptions.Protocol = OtlpExportProtocol.HttpProtobuf;
        })
        .SetResourceBuilder(ResourceBuilder.CreateDefault().AddAttributes(new Dictionary<string, object>
        {
            ["project"] = "test-project",
            ["subproject"] = "test-subproject",
            ["application"] = "test-application",
            ["environment"] = "staging",
        }))
    )
);
var logger = loggerFactory.CreateLogger<Program>();

logger.LogError(new Exception("exception"), "test");
```

The application continues to run even if the configuration is incorrect. For example, I can specify a "https://blahblahblah" as a endpoint and the application will not give any hint that something is wrong. 

So I can't figure out why logs are not sent to the specified endpoint. How to enable "fail on non-success requests" **or/and** enable "logging of logging" **or/and** where can I debug a requests sending?

**Comment by cijothomas** [comment]
If the url itself is wrong, it makes sense to throw exception. But if the endpoint is not reachable at startup, it is debatable whether to throw exception and fail right away, or keep "hope" that the endpoint might be available eventually....

The best bet right now is to rely on [self-diagnostics](https://github.com/open-telemetry/opentelemetry-dotnet?tab=readme-ov-file#troubleshooting) which should tell you exactly why export is failing, if it fails...
